### PR TITLE
send warnings with info in error format

### PR DESCRIPTION
**Pull request by iCollin**
Fixes successful WARNINGS info not being passed to mobile

This PR is **ready** for review.

### Summary
success message schema does not contain info

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: ffw/NavigationRPC.js [hunk 1]
```diff
@@ -502,6 +502,11 @@ FFW.Navigation = FFW.RPCObserver.create(
         return;
       }
 
+      // (&& !params) - params take precedent over info
```
Comment (AKalinich-Luxoft) [inline ffw/NavigationRPC.js:505]: @iCollin propose to add a TODO statement here

File: ffw/NavigationRPC.js [hunk 2]
```diff
@@ -502,6 +502,11 @@ FFW.Navigation = FFW.RPCObserver.create(
...
+      if (info && resultCode === SDL.SDLModel.data.resultCode.WARNINGS) {
```
Comment (AKalinich-Luxoft) [inline ffw/NavigationRPC.js:506]: Actually it is applicable for any successful result code
```suggestion
      if (info && is_successful_code) {
```

**Review by AKalinich-Luxoft — APPROVED** [review]

File: ffw/BasicCommunicationRPC.js [hunk 3]
```diff
@@ -990,8 +987,12 @@ FFW.BasicCommunication = FFW.RPCObserver
           return;
         }
 
+        // TODO params take precedent over info:
```
Comment (JackLivio) [inline ffw/BasicCommunicationRPC.js:990]: Should this addition be included in this pr?

File: ffw/BasicCommunicationRPC.js [hunk 4]
```diff
@@ -1002,9 +1003,10 @@ FFW.BasicCommunication = FFW.RPCObserver
             }
           };
 
-          if (info) {
-            JSONMessage.result.info = info;
-          }
+          // SDL success response schema does not have info param as of 7.0
```
Comment (JackLivio) [inline ffw/BasicCommunicationRPC.js:1006]: Remove commented lines

File: ffw/NavigationRPC.js [hunk 5]
```diff
@@ -460,22 +460,19 @@ FFW.Navigation = FFW.RPCObserver.create(
...
-        var JSONMessage = {
-          'jsonrpc': '2.0',
-          'id': id,
-          'error': {
-            'code': resultCode, // type (enum) from SDL protocol
-            'message': message,
-            'data': {
-              'method': method
-            }
+      // send repsonse
```
Comment (JackLivio) [inline ffw/NavigationRPC.js:463]:
```suggestion
      // send response
```

File: ffw/NavigationRPC.js [hunk 6]
```diff
@@ -514,9 +515,10 @@ FFW.Navigation = FFW.RPCObserver.create(
           }
         };
 
-        if(info != null) {
-          JSONMessage.result.info = info;
-        }
+        // SDL success response format does not have info param as of 7.0
+        //if(info != null) {
```
Comment (JackLivio) [inline ffw/NavigationRPC.js:519]: remove commented lines

File: ffw/BasicCommunicationRPC.js (hunk 3 above)
Comment (iCollin) [reply ffw/BasicCommunicationRPC.js:990]: probably, it is implemented in https://github.com/smartdevicelink/sdl_hmi/pull/427

@AKalinich-Luxoft do you mind targeting this PR with your #427 and I will merge it in?

Comment (AKalinich-Luxoft) [reply ffw/BasicCommunicationRPC.js:990]: @iCollin I am okay with that

**Review by JackLivio — APPROVED** [review]

**Pull request merged**